build.sbt

```
@@ -644,7 +648,8 @@ lazy val cross = project
   .settings(
     testSettings,
     sharedSettings,
-    crossScalaVersions := V.nonDeprecatedScalaVersions
+    crossScalaVersions :=
+      V.nonDeprecatedScalaVersions ++ V.nigthtlyScala3Versions.lastOption.toList
```

Could we only run this on a separate job and remove the check from required for a merge? I will probably need to do it manually later on, but let's maybe have this run separately. This could be done via special env variable maybe?
```suggestion
    crossScalaVersions := {
         if (System.getenv("TEST_NIGHTLY") != null)  V.nightlyScala3Versions.lastOption.toList 
         else V.nonDeprecatedScalaVersions
     }
```
and add that variable to a separate workflow. What do you think?

---

Good idea!
Will rework it a bit differently to call just a `++$LATEST_NIGHTLY cross/test` from a separate job